### PR TITLE
adjust e2e pod resources

### PR DESCRIPTION
**Pull request by dragonly**
### What problem does this PR solve?

adjust e2e pod resources to avoid eviction during job run.

### What is changed and how does it work?

adjust ephemeral-storage/cpu/memory of e2e pods

### Code changes

- [ ] Has Go code change
- [x] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [x] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
NONE
```

### CODE REVIEW

File: ci/pull_e2e_kind.groovy [hunk 1]
```diff
@@ -251,12 +251,12 @@ try {
...
             requests: [
-                cpu: "4",
-                memory: "4G"
+                cpu: "6",
+                memory: "10Gi"
             ],
             limits: [
                 cpu: "8",
-                memory: "32G"
+                memory: "16Gi"
```
Comment (DanielZhangQD) [inline ci/pull_e2e_kind.groovy:259]: We can keep 32G for this.

Comment (DanielZhangQD) [inline ci/pull_e2e_kind.groovy:259]:
```suggestion
                memory: "32Gi"
```

File: ci/pull_e2e_kind.groovy [hunk 2]
```diff
@@ -365,10 +365,8 @@ try {
                     }
                 }
             }
-        }
-
+            }
```
Comment (DanielZhangQD) [inline ci/pull_e2e_kind.groovy:368]: The indent here is incorrect.

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Comment by DanielZhangQD** [comment]
@dragonly The Pod is evicted again, it seems that the ephemeral storage change does not take effect:
```
Status:             Failed
Reason:             Evicted
Message:            Pod ephemeral local storage usage exceeds the total limit of containers 70Gi.
```

**Comment by DanielZhangQD** [comment]
The resources are not changed either:
```
    Limits:
      cpu:                8
      ephemeral-storage:  70Gi
      memory:             8Gi
    Requests:
      cpu:                4
      ephemeral-storage:  70Gi
      memory:             4Gi
```

**Comment by dragonly** [comment]
> @dragonly The Pod is evicted again, it seems that the ephemeral storage change does not take effect:
> 
> ```
> Status:             Failed
> Reason:             Evicted
> Message:            Pod ephemeral local storage usage exceeds the total limit of containers 70Gi.
> ```

@DanielZhangQD That is because the jenkins job always run groovy pipeline defined in master branch. This will only take effect when merged into master branch
image

**Pull request merged**